src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -2790,7 +2790,11 @@ private void CompileTypeEqualExpression(Expression expr)
             var node = (TypeBinaryExpression)expr;
 
             Compile(node.Expression);
-            if (node.TypeOperand.GetTypeInfo().IsGenericType && node.TypeOperand.GetGenericTypeDefinition() == typeof(Nullable<>))
+            if (node.Expression.Type == typeof(void))
+            {
+                _instructions.EmitLoad(node.TypeOperand == node.Expression.Type, typeof(bool));
```

Comment:
Well, this is a slightly different case to that, though clearly related. Here we've got a value being popped from the stack after an expression that didn't add one, with a resulting invalid stack state.